app/src/main/java/com/hadiyarajesh/notex/ui/note/add/AddNoteScreen.kt

```diff
@@ -1,31 +1,190 @@
 @Composable
 fun AddNoteScreen(
-    navController: NavController
+    navController: NavController, addNotesViewModel: AddNotesViewModel, noteId: Long = -1
 ) {
-    Scaffold { innerPadding ->
+    val state = remember {
+        mutableStateOf(NoteState())
+    }
+
+    val noteBackground = remember {
+        Animatable(
+            Color(state.value.color.value)
+        )
+    }
+    LaunchedEffect(key1 = true) {
+        if (noteId.compareTo(-1) != 0) {
+            val note = addNotesViewModel.getNote(noteId)
+            state.value.apply {
+                title.value = note.title ?: ""
+                noteDesc.value = note.content ?: ""
+                color.value = Color(android.graphics.Color.parseColor(note.color)).toArgb()
+            }
+
+            noteBackground.animateTo(
+                targetValue = Color(state.value.color.value),
+                animationSpec = tween(
+                    durationMillis = 400
+                )
+            )
+
+        }
+    }
+
+    BackHandler {
+        addNotesViewModel.saveNote(state.value, noteId)
+        navController.navigateUp()
+    }
+
+    Scaffold(bottomBar = {
+        BottomAppBar(containerColor = noteBackground.value, modifier = Modifier.height(40.dp)) {
+            ShowColor(state = state, noteBackground)
+        }
+    }) { innerPadding ->
         Column(
             modifier = Modifier
                 .fillMaxSize()
+                .background(noteBackground.value)
                 .padding(innerPadding),
             horizontalAlignment = Alignment.CenterHorizontally,
-            verticalArrangement = Arrangement.Center
-        ) {
-            Text(text = "Add some notes here")
+            verticalArrangement = Arrangement.Top,
+
+            ) {
+            TopAppBar(navController, addNotesViewModel, state, noteId)
+            Column(
+                modifier = Modifier.padding(
+                    start = 16.dp,
+                    end = 16.dp,
+                    top = 10.dp,
+                    bottom = 1.dp
+                )
+            ) {
+                BorderLessTextField(
+                    text = state.value.title.value,
+                    hint = "Title",
+                    onValueChange = {
+                        state.value.title.value = it
+                    },
+                    maxLines = 1,
+                    textStyle = MaterialTheme.typography.headlineMedium
+                )
+                VerticalSpacer(size = 16)
+                BorderLessTextField(
+                    text = state.value.noteDesc.value,
+                    hint = "write note here",
+                    onValueChange = {
+                        state.value.noteDesc.value = it
+                    },
+                    textStyle = MaterialTheme.typography.headlineSmall
+                )
+            }
+        }
+    }
+}
+
+
+@Composable
+private fun TopAppBar(
+    navController: NavController,
```

Comment:
Hoist this click event to parent composable

Reply:
done